examples/nsl_pk/DY.Example.NSL.Protocol.Stateful.Proof.fst

```diff
@@ -54,15 +54,15 @@ let event_predicate_nsl: event_predicate nsl_event =
     match e with
     | Initiate1 alice bob n_a -> (
       prin == alice /\
-      get_label n_a == join (principal_label alice) (principal_label bob) /\
-      0 < DY.Core.Trace.Type.length tr /\
-      rand_generated_at tr (DY.Core.Trace.Type.length tr - 1) n_a
+      is_secret (join (principal_label alice) (principal_label bob)) tr n_a /\
+      0 < DY.Core.Trace.Base.length tr /\
+      rand_generated_at tr (DY.Core.Trace.Base.length tr - 1) n_a
```

Comment:
This also seems like a common enough condition for events (as we're handling things right now, anyway) to also encapsulate in a predicate.

Reply:
We could do that. I think this particular technique is quite specific to NSL, so we could also define this predicate in the NSL example.

Reply:
Moving the discussion in #61.